jwst/wfss_contam/observations.py

```diff
@@ -0,0 +1,545 @@
+    """
+
+    def __init__(self, xi, yi, **kwargs):
+        self.xi = xi
+        self.yi = yi
+        self.args = kwargs
+        self.f = interp1d(xi, yi, **kwargs)
+
+    def __call__(self, xnew):
+        return self.f(xnew)
+
+    def __getstate__(self):
+        return self.xi, self.yi, self.args
+
+    def __setstate__(self, state):
+        self.f = interp1d(state[0], state[1], **state[2])
+
+
+class observation():
+    # This class defines an actual observation. It is tied to a single grism image.
+
+    def __init__(self, direct_images, segmap_model, grism_wcs, wr_ref, filter, config,
+                 mod="A", order=1, max_split=100, SED_file=None, extrapolate_SED=False, max_cpu=8,
+                 ID=0, SBE_save=None, boundaries=[], renormalize=True):
+
+        """
+        direct_images: List of file name(s) containing direct imaging data
+        segmentation_data: a data array of the same size as the direct image(s), containing 0's in
+            pixels with no source signal and an integer source number for pixels with signal
+        config: Full path name of a NIRCam GRISMCONF configuration file
+        mod: NIRCam Module: A or B
+        order: The spectral order to simulate, +1 or +2 for NIRCAM
+        max_split: Number of chunks to compute instead of trying everything at once.
+        SED_file: Name of HDF5 file containing datasets matching the ID in the segmentation file
+                  and each consisting of a [[lambda],[flux]] array.
+        extrapolate_SED:
+        max_cpu:
+        ID:
+        SBE_save: If set to a path, HDF5 containing simulated stamps for all obsjects will be saved.
+        boundaries: a tuple containing the coordinates of the FOV within the larger seed image.
+        renormalize:
+        """
+
+        # This loads all the info from the configuration file for this grism mode;
+        # things like the name of the POM mask file, the dispersion coeffs, etc.
+        self.seg_wcs = segmap_model.meta.wcs
+        self.grism_wcs = grism_wcs
+        self.ID = ID
+        self.IDs = []
+        self.dir_image_names = direct_images
+        self.seg = segmap_model.data
+        # self.dims = np.shape(self.seg)
+        self.dims = [2048, 2048]
+        self.filter = filter
+        self.order = order
+        self.SED_file = SED_file   # should always be NONE for baseline pipeline (use flat SED)
+        self.SBE_save = SBE_save   # may want to have a new product in which simulated spectra are saved (NOT HDF5 format!)
+        self.max_cpu = max_cpu
+        self.cache = False
+        self.POM_mask = None
+        self.POM_mask01 = None
+        self.renormalize = renormalize
+
+        # Get the wavelength range for the grism image
+        wavelength_range = wr_ref.get_wfss_wavelength_range(self.filter, [self.order])
+        self.wmin = wavelength_range[self.order][0]
+        self.wmax = wavelength_range[self.order][1]
+        print("wmin, wmax:", self.wmin, self.wmax)
```

Comment:
`self.wmin, self.wmax = wavelength_range[self.order]`

Remove print

Reply:
Should those be saved by assign_wcs in in `wcsinfo[wave_start/end]`

Reply:
They could be and that would make it easier here, so that the wavelengthrange ref file doesn't need to be loaded again just to get those 2 values. In general in might be useful to have `assign_wcs` save/populate `WAVSTART` and `WAVEND` for all non-IFU spectroscopic modes, instead of just NIRSpec. Even if not used in downstream pipeline, it could be useful for DAT and users.